File: client/momo/lib/app/model/user/user_update_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:momo/app/model/common/code_name_pair.dart';

part 'user_update_response.g.dart';
part 'user_update_response.freezed.dart';

@freezed
class UserUpdateResponse with _$UserUpdateResponse {
  factory UserUpdateResponse({
    required String nickname,
    required CodeNamePair city,
    required String district,
    required String university,
    String? imageUrl,
  }) = _UserUpdateReponse;

  factory UserUpdateResponse.fromJson(Map<String, dynamic> json) =>
      _$UserUpdateResponseFromJson(json);
}

File: server/momo-api/src/main/java/com/momo/api/user/dto/response/UserUpdateResponse.java
package com.momo.api.user.dto.response;

import com.momo.domain.common.dto.EnumResponse;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class UserUpdateResponse {

    private String nickname;
    private String university;
    private EnumResponse city;
    private String district;

    @Builder
    public UserUpdateResponse(String nickname, String university, EnumResponse city, String district) {
        this.nickname = nickname;
        this.university = university;
        this.city = city;
        this.district = district;
    }
}
